sphinx_immaterial/content_tabs.py

```diff
@@ -0,0 +1,207 @@
+        return [tab_set]
+
+
+class MaterialTabItemDirective(SphinxDirective):
+    """A single tab item in a tab set.
+    Note: This directive generates a single container,
+    for the label and content::
+        <container design_component="tab-item" has_title=True>
+            <rubric>
+                ...title nodes
+            <container design_component="tab-content">
+                ...content nodes
+    This allows for a default rendering in non-HTML outputs.
+    The ``MaterialTabSetHtmlTransform`` then transforms this container
+    into the HTML specific structure.
+    """
+
+    required_arguments = 1  # the tab label is the first argument
+    final_argument_whitespace = True
+    has_content = True
+    option_spec = {
+        "class": directives.class_option,
+    }
+
+    def run(self) -> List[nodes.Node]:
+        """Run the directive."""
+        self.assert_has_content()
+        if not is_md_tab_type(self.state_machine.node, "md-tab-set"):
+            LOGGER.warning(
+                "The parent of a 'md-tab-item' should be a 'md-tab-set'",
+                location=(self.env.docname, self.lineno),
+            )
+        tab_item = nodes.container(
+            "", is_div=True, type="md-tab-item", classes=["tabbed-block"]
+        )
+
+        # add tab label
+        textnodes, _ = self.state.inline_text(self.arguments[0], self.lineno)
+        tab_label = nodes.rubric(
+            self.arguments[0],
+            *textnodes,
+            classes=["tabbed-label"],
+        )
+        self.add_name(tab_label)
+        tab_item += tab_label
+
+        # add tab content
+        tab_content = nodes.container(
+            "",
+            is_div=True,
+            type="",
+            classes=["tabbed-block"] + self.options.get("class", []),
+        )
+        self.state.nested_parse(self.content, self.content_offset, tab_content)
+        tab_item += tab_content
+
+        return [tab_item]
+
+
+class content_tab_input(nodes.Element, nodes.General):
+    pass
+
+
+class content_tab_label(nodes.TextElement, nodes.General):
+    pass
+
+
+def visit_tab_input(self, node):
+    attributes = {"ids": [node["id"]], "type": node["type"], "name": node["set_id"]}
+    if node["checked"]:
+        attributes["checked"] = "checked"
+    self.body.append(self.starttag(node, "input", **attributes))
+
+
+def depart_tab_input(self, node):
+    self.body.append("</input>")
+
+
+def visit_tab_label(self, node):
+    attributes = {"for": node["input_id"]}
+    self.body.append(self.starttag(node, "label", **attributes))
+
+
+def depart_tab_label(self, node):
+    self.body.append("</label>")
+
+
+def visit_tab_set(self: HTMLTranslator, node: content_tab_set):
+    # increment tab set counter
+    self.tab_set_count = getattr(self, "tab_set_count", 0) + 1
+
+    # configure tab set's div attributes
+    tab_set_identity = "__tabbed_" + str(self.tab_set_count)
+    attributes = {"data-tabs": str(self.tab_set_count) + node["data-tabs"]}
```

Comment:
Potentially rather than have an extra data-tabs attribute you could just compute that here from `len(node.children)`

Reply:
I've found that there are multiple ways to count the children. I have no objection to this suggestion. I just set the attribute since I had to traverse the child nodes (in `Material TabSetDirective.run()`) to verify they were of `md-tab-item` directives (so I tallied the number of children there).